sql/moz-fx-data-shared-prod/telemetry_derived/sponsored_tiles_clients_daily_v1/query.sql

```diff
@@ -112,22 +112,38 @@ ios_events AS (
 ),
 ios_clients AS (
   SELECT
-    date(submission_timestamp) AS submission_date,
-    client_info.client_id,
-    `mozfun.norm.browser_version_info`(client_info.app_display_version) AS browser_version_info,
-    mozfun.glean.legacy_compatible_experiments(ping_info.experiments) AS experiments,
-    normalized_country_code AS country,
-    client_info.locale,
-    normalized_channel,
-    normalized_os_version,
-    sample_id
+    ARRAY_AGG(DATE(submission_timestamp) ORDER BY submission_timestamp DESC)[
+      OFFSET(0)
+    ] AS submission_date,
```

Comment:
You could replace all these `ARRAY_AGG` statements with

```
FIRST_VALUE(x) OVER (PARTITION BY client_id ORDER BY submission_timestamp DESC)
```

You'd also have to throw a `SELECT DISTINCT` at the top, which is too bad.

Maybe it's just personal preference, but that feels to me more like the preferred way to do it.